apps/admin/frontend/src/utils/reporting.ts

```diff
@@ -103,39 +108,79 @@ export function generateTitleForReport({
     if (scannerId) {
       return ok(`Scanner ${scannerId} ${reportType} Report`);
     }
-  }
 
-  if (reportRank === 2) {
-    if (precinctId && votingMethod) {
+    if (partyId) {
       return ok(
-        `${getPrecinctById(electionDefinition, precinctId).name} ${
-          VOTING_METHOD_LABELS[votingMethod]
-        } Ballot ${reportType} Report`
+        `${
+          getPartyById(electionDefinition, partyId).fullName
+        } ${reportType} Report`
       );
     }
+  }
 
-    if (ballotStyleId && votingMethod) {
-      return ok(
-        `Ballot Style ${ballotStyleId} ${VOTING_METHOD_LABELS[votingMethod]} Ballot ${reportType} Report`
-      );
+  if (reportRank === 2) {
+    // Party + Other
+    if (partyId) {
```

Comment:
These cases were not relevant before because the tally reports cannot have a `partyId`, but the ballot count reports can.